src/System.Json/src/System/Json/JsonValue.cs

```diff
@@ -122,7 +124,10 @@ public virtual void Save(Stream stream)
                 throw new ArgumentNullException(nameof(stream));
             }
 
-            Save(new StreamWriter(stream));
+            using (StreamWriter writer = new StreamWriter(stream, s_encoding, 1024, true))
```

Comment:
why do you pass the last parameter as true here? shouldn't pass it at all or pass it false?

Reply:
I'm not sure I understand the question, why should calling this method close the stream? It doesn't mention doing so in the documentation, nor do any of its overriding methods close it.

Reply:
the last parameter of ```StreamWriter(stream, s_encoding, 1024, true) ``` is documented as 

```
leaveOpen
Type: System.Boolean
true to leave the stream open after the StreamWriter object is disposed; otherwise, false.

https://msdn.microsoft.com/en-us/library/gg712853(v=vs.110).aspx
```

and you are passing true value. so my question is, why you need to keep the stream opened after disposing of StreamWriterObject? 
Also, the original code used ```StreamWriter(stream)``` which eventually will call the other constructor with parameters like ```C# StreamWriter(stream, s_encoding, 1024, false)```. note that it is passing false to force closing the stream.

Reply:
The `StreamWriter` in the original code was never disposed, so it made no difference that it defaulted to closing the stream. It also never flushed, meaning only part of the output ended up in the stream (or none of it, if your json object was small enough...).

Reply:
> The StreamWriter in the original code was never disposed

Is this true? wouldn't at some point the GC will kick in and finalize this object?

Reply:
First of all neither `StreamWriter` nor its base class `TextWriter` has a finalizer. Which is correct; neither of these classes owns unmanaged resources. The GC doesn't see `IDisposable.Dispose` as special, and will not call it for you.

Secondly even if `StreamWriter` had a finalizer which would call `Dispose`, there is no guarantee that the GC *ever* runs (and, more reasonably, there is no guarantee that the GC would run before you start performing actions that rely on the json-data being written to the stream), so we wouldn't be able to rely on this behaviour anyway.

Thirdly, if `JsonValue.Save(Stream)` would close the stream, it would be impossible to use for most json values. For example, `new JsonArray(new JsonObject(), new JsonObject()).Save(new MemoryStream())` would throw an exception (either `NotSupportedException` or `ObjectDisposedException`, I predict), because `JsonArray` calls `JsonValue.Save(stream)` on its elements, so serializing the first `JsonObject` would close the stream and dispose the stream,

In short, I think we really need to pass `true` as the final constructor parameter :)

Reply:
ok. make sense then. thanks for the clarification.